src/librustc_target/abi/call/mips64.rs

```diff
@@ -88,6 +88,7 @@ where
     let mut prefix_index = 0;
 
     match arg.layout.fields {
+        abi::FieldsShape::Primitive => unreachable!(),
```

Comment:
I think you can move the code from the `if !arg.layout.is_aggregate()` above into this arm and move the `let dl` into the `Arbitrary` arm.